### DISCUSSION TITLE
Problemi installazione firmware RCP su ZBDongle-E (v2)

### DISCUSSION THREAD

**Discussion by lnrd00** [indomus/forum#2904] [Hardware & Firmware]
Salve, scrivo per avere chiarimenti. 
Ho seguito la guida inDomus per l'[aggiornamento del firmware del coordinator ZBDongle-E](https://indomus.it/guide/riprogrammare-il-firmware-di-itead-sonoff-zbdongle-e-zigbee-3-0-usb-dongle-plus-v2/#webapp).
Dopo aver eseguito con successo l'installazione del firmware RCP (ho scelto di installare quello RCP per predisporre la mia domotica anche all'utilizzo di Thread) attraverso l'applicazione su browser, ho riconnesso l'antenna Zigbee al Raspberry a cui era collegata.
Però adesso, se provo a connettermi alla pagina web di zigbee2mqtt, attraverso `http://XXX.XXX.X.XXX:8080` ottengo la seguente pagina:
<img width="662" alt="Schermata 2024-02-10 alle 16 08 37" src="https://github.com/indomus/forum/assets/154708323/7168cc53-1293-44ce-ab5c-f06ea2e3af63">
E soprattutto, questi sono i logs del container Zigbee2MQTT:

```
Zigbee2MQTT:info  2024-02-10 15:59:47: Starting zigbee-herdsman (0.30.0)
Error: Reset error: Error: {"sequence":-1} after 10000ms
    at /app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:299:23
    at Queue.executeNext (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:32:32)
Using '/app/data' as data directory
```
Preciso che:

- Avevo stoppato il container zigbee2mqtt prima della rimozione dell'antenna e l'ho fatto ripartire dopo aver resinserito l'antenna
- ho installato tutti i driver necessari per l'aggiornamento firmware, che comunque ripeto sembrerebbe essere andato a buon fine.
- Ho installato l'ultimo firmware disponibile [al seguente link](https://github.com/darkxst/silabs-firmware-builder/tree/main/firmware_builds/zbdonglee), ovvero [questo firmware](https://github.com/darkxst/silabs-firmware-builder/blob/main/firmware_builds/zbdonglee/rcp-uart-802154-v4.4.0-zbdonglee-460800.gbl)

È normale sia così a seguito della riprogrammazione, e devo compiere ancora qualche passaggio, oppure ho fatto qualche errore/c'è stato qualche problema nell'installazione del firmware?

Grazie in anticipo!!

**AGGIORNAMENTO**
Ho provato ad installare il firmware consigliato dall'app browser, che credo sia la versione `4.3.1`, ma continua a dare lo stesso errore.

**Comment by snakuzzo** [comment]
Ma hai avviato l'addon/applicazione Silabs Multiprotocol ?

**Reply by lnrd00** [reply]
Ho provato a istanziare Silicon Labs Multiprocol su Docker Compose, ma digitando `docker logs multipan` ricevo il seguente errore:

```
2024-02-10 16:41:38 [INFO] Starting cpcd...
WARNING in function 'main' in file /usr/src/cpc-daemon/main.c at line #186 : Running CPCd as 'root' is not recommended. Proceed at your own risk.
FATAL in function 'protocol_version_check' in file /usr/src/cpc-daemon/server_core/server_core.c at line #729 : Secondary Protocol v5 doesn't match CPCd Protocol v4
[16:41:38:997688] Info : [CPCd v4.3.1.0] [Library API v3] [RCP Protocol v4]
[16:41:38:997746] Info : Git commit: 133b29678b3d0bc7578e098d2f46b4d5bcd2ebb4 / branch: 
[16:41:38:997754] Info : Sources hash: ff8300587e7e4ab1def7a89a272c0baef32f9eb3bff9b0ba06b94e655d652367
[16:41:38:997761] WARNING : In function 'main' in file /usr/src/cpc-daemon/main.c at line #186 : Running CPCd as 'root' is not recommended. Proceed at your own risk.
[16:41:38:997778] Info : Reading cli arguments
[16:41:38:997785] Info : /usr/local/bin/cpcd 
[16:41:38:998188] Info : Reading configuration
[16:41:38:998200] Info :   file_path = /usr/local/etc/cpcd.conf
[16:41:38:998206] Info :   instance_name = cpcd_0
[16:41:38:998211] Info :   socket_folder = /dev/shm
[16:41:38:998215] Info :   operation_mode = MODE_NORMAL
[16:41:38:998221] Info :   use_encryption = false
[16:41:38:998225] Info :   binding_key_file = /root/.cpcd/binding.key
[16:41:38:998230] Info :   stdout_tracing = false
[16:41:38:998235] Info :   file_tracing = false
[16:41:38:998240] Info :   lttng_tracing = false
[16:41:38:998244] Info :   enable_frame_trace = false
[16:41:38:998249] Info :   traces_folder = /dev/shm/cpcd-traces
[16:41:38:998254] Info :   bus = UART
[16:41:38:998259] Info :   uart_baudrate = 460800
[16:41:38:998264] Info :   uart_hardflow = false
[16:41:38:998269] Info :   uart_file = /dev/ttyACM0
[16:41:38:998273] Info :   fu_recovery_pins_enabled = false
[16:41:38:998278] Info :   fu_connect_to_bootloader = false
[16:41:38:998283] Info :   fu_enter_bootloader = false
[16:41:38:998288] Info :   restart_cpcd = false
[16:41:38:998293] Info :   application_version_validation = false
[16:41:38:998298] Info :   print_secondary_versions_and_exit = false
[16:41:38:998303] Info :   use_noop_keep_alive = false
[16:41:38:998307] Info :   reset_sequence = true
[16:41:38:998312] Info :   stats_interval = 0
[16:41:38:998317] Info :   rlimit_nofile = 2000
[16:41:38:998322] Info : ENCRYPTION IS DISABLED 
[16:41:38:998327] Info : Starting daemon in normal mode
[16:41:39:009539] Info : Connecting to Secondary...
[16:41:39:087155] Info : RX capability is 256 bytes
[16:41:39:087180] Info : Connected to Secondary
[16:41:39:090150] Info : Secondary Protocol v5
[16:41:39:090158] *** ASSERT *** : FATAL in function 'protocol_version_check' in file /usr/src/cpc-daemon/server_core/server_core.c at line #729 : Secondary Protocol v5 doesn't match CPCd Protocol v4
[16:41:40:090560] Info : Daemon exiting with status EXIT_FAILURE
Logger buffer size = 28672, highwater mark = 2442 : 8.52%. Lost logs : 0
2024-02-10 16:41:40 [INFO] CPC ended with exit code 1 (signal 0)...
```

Il dubbio che avevo inizialmente sollevato in questa discussione risale a **prima** che io provassi a istanizare Silicon Labs Multiprotocol, ma adesso ho questo problema. 
Tra i gli stati dei vari container, questo di Silabs è definito come "unhealthy".

Di seguito inserisco il codice che ho aggiunto al file `compose.yaml` per la configurazione del container:

```
  multipan:
    container_name: multipan
    image: b2un0/silabs-multipan-docker
    cap_add:
      - SYS_ADMIN
      - NET_ADMIN
    restart: always
    privileged: true
    network_mode: host
    devices:
      - /dev/ttyACM0
    volumes:
      - ./data/:/data
    environment:
      # OCPD environment variables
      DEVICE: "/dev/ttyACM0"
      BAUDRATE: "460800"
      CPCD_TRACE: "false"
      FLOW_CONTROL: "false"
      NETWORK_DEVICES: 0
      # OTBR-AGENT environment variables
      OTBR_ENABLE: 1
      BACKBONE_IF: "eth0"
      OTBR_LOG_LEVEL: "debug"
      OTB_FIREWALL: 1
      OTBR_REST_LISTEN_PORT: "8081"
      # OTBR-WEB environment variables
      OTBR_WEB_PORT: "8086"
      # SOCAT environment variables
      NETWORK_DEVICE: ""
      # ZIGBEED environment variables
      EZSP_LISTEN_PORT: "9999"
      # UNIVERSAL FLASHER variables
      AUTOFLASH_FIRMWARE: 0
      FIRMWARE: "86"
```

Il codice è uguale a quello consigliato sulla [guida inDomus](https://indomus.it/guide/come-installare-e-configurare-silicon-labs-multiprotocol-per-gestire-coordinator-multipan-rcp/#port), dove ho soltanto inserito i miei `BACKBONE_IF` e `DEVICE`/`devices`.

**Reply by snakuzzo** [reply]
ok...però io dico...
come cavolo ti aspetti che funzioni zigbee2mqtt (dove comunque dovrai cambiare la configurazione della seriale) senza avere siliabs multiprotocol funzionante ?
Hai almeno capito architetturalmente come sono legati i vari software e l'hardware ?

Te lo chiedo perché altrimenti vai avanti con i copia-incolla senza capire nulla

**Reply by lnrd00** [reply]
La configurazione della seriale credo di averla cambiata correttamente. 

1. Sul `configuration.yaml` di `zigbee2mqtt`:

```
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://ip-del-broker-mqtt
  user: xxxxxx
  password: xxxxxx
  version: 5
  include_device_information: true
serial:
  port: tcp://core-silabs-multiprotocol:9999
  adapter: ezsp
advanced:
  log_directory: /var/log
experimental: {}
frontend:
  port: 8080
ota:
  update_check_interval: 10
disable_automatic_update_check: false
```

2. Mentre sul file `compose.yaml` ho rimosso il campo `Devices` dalla configurazione di `zigbee2mqtt`

**Reply by snakuzzo** [reply]
Prima di verificare questo dovresti assicurarti che funzioni silabs. Senza quello, tutte le altre prove sono inutili

**Reply by lnrd00** [reply]
silabs non sta evidentemente funzionando, ma non so cosa fare per risolvere il problema che mi viene dato nei logs, dove sembra centrale l’errore

`*** ASSERT *** : FATAL in function 'protocol_version_check' in file /usr/src/cpc-daemon/server_core/server_core.c at line #729 : Secondary Protocol v5 doesn't match CPCd Protocol v4`

**Reply by snakuzzo** [reply]
Ricapitolando, il terzetto funzionante (come abbiamo scritto in chat e sul sito e che ho appena riprovato)....

- Zigbee2mqtt 1.35.3 (ultima release stable)
- Silabs Multiprotocol 2.4.4 (ultima release) - qui devi capire se la versione ultima del container che usi sia effettivamente basata sul sorgente della 2.4.4 dell'addon
- Firmeware 4.3.1 (il 4.4.0 NON va utilizzato, come anche indicato nella nota di silabs multiprotocol [QUI](https://github.com/home-assistant/addons/blob/master/silabs-multiprotocol/CHANGELOG.md#244))

Inoltre, quando risolverai il problema legato allo start di silabs, nella tua configurazione di zigbee2mqtt, devi assicurarti che il container di silabs venga effettivamente risolto con il nome che hai specificato (ovvero `core-silabs-multiprotocol`)

Infine, per il tuo errore...prova a guardare [QUI](https://github.com/home-assistant/addons/issues/3412)

**Reply by lnrd00** [reply]
Ti aggiorno sulla situazione. Ho corretto la versione del firmware dell'antenna coordinator, riportandola alla 4.3.1, e adesso tutto sembrerebbe essere funzionante. I logs di silabs e di zigbee2mqtt sembrano non evidenziare errori, pure la pagina web di zigbee2mqtt è tornata operativa. Inoltre, Home Assistant mi mostra questo:
<img width="356" alt="Schermata 2024-02-12 alle 00 24 00" src="https://github.com/indomus/forum/assets/154708323/8f9b7d92-f701-49b0-b9bb-8048caf05de0">
Che sembrerebbe essere di buon auspicio.

- > Zigbee2mqtt 1.35.3 (ultima release stable)

- >Silabs Multiprotocol 2.4.4 (ultima release) - qui devi capire se la versione ultima del container che usi sia effettivamente basata sul sorgente della 2.4.4 dell'addon

Non sono riuscito a verificare queste due informazioni, ma ho istanziato silabs giusto ieri sera dunque credo proprio che sia aggiornato all'ultima versione. L'unica è zigbee2mqtt, che non sono riuscito a capire a che versione sia rimasta, ma mi verrebbe da pensare che sia una release stabile dato che adesso è tutto funzionante.

**Reply by snakuzzo** [reply]
Praticamente la cosa più facile, vai sulla dashboard di Zigbee2mqtt o nei log e vedi che versione è.
Puoi anche vederlo dalle immagini docker che stai usando
